### PR TITLE
[dev2] IntelCC reference

### PR DESCRIPTION
**Pull request by franramirez688**

### CODE REVIEW

File: reference/tools/intel.rst [hunk 1]
```diff
@@ -1,6 +1,148 @@
...
 IntelCC
-=======
+-------
+
+This tool helps you to manage the new Intel oneAPI `DPC++/C++ <https://software.intel.com/content/www/us/en/develop/documentation/oneapi-dpcpp-cpp-compiler-dev-guide-and-reference/top.html>`_ and
+`Classic <https://software.intel.com/content/www/us/en/develop/documentation/cpp-compiler-developer-guide-and-reference/top.html>`_ ecosystem in Conan.
+
+.. warning::
+
+    This generator is **experimental** and subject to breaking changes.
```
Comment (franramirez688) [inline reference/tools/intel.rst:14]: I think this generator is barely used so I prefer to keep it.

File: reference/tools/intel.rst [hunk 2]
```diff
@@ -1,6 +1,148 @@
...
+    :members:
+
+
+conf
+++++
+
+``IntelCC`` uses these :ref:`configuration entries <reference_config_files_global_conf>`:
 
-TBD
+- ``tools.intel:installation_path``: **(required)** argument to tells Conan the installation path, if it's not defined,
```
Comment (SSE4) [inline reference/tools/intel.rst:145]:
```suggestion
- ``tools.intel:installation_path``: **(required)** argument to tell Conan the installation path, if it's not defined,
```

File: reference/tools/intel.rst [hunk 3]
```diff
@@ -1,6 +1,148 @@
...
+  Conan will try to find it out automatically.
+- ``tools.intel:setvars_args``: **(optional)** it is used to pass whatever we want as arguments to our `setvars.sh|bat` file.
```
Comment (SSE4) [inline reference/tools/intel.rst:147]: it's a bit unclear. can you put some link here? `whatever we want` doesn't say much.

File: reference/tools/intel.rst [hunk 4]
```diff
@@ -1,6 +1,148 @@
...
+    ...
+    compiler=intel-cc
+    compiler.mode=dpcpp
+    compiler.version=2021.3
+    compiler.libcxx=libstdc++
+    build_type=Release
+
+    [buildenv]
+    CC=dpcpp
+    CXX=dpcpp
```
Comment (memsharded) [inline reference/tools/intel.rst:57]: Does this work on Windows too? Maybe something to discuss in the future if this default could/should be automated.

**Review by memsharded — APPROVED** [review]

File: reference/tools/intel.rst (hunk 4 above)
Comment (franramirez688) [reply reference/tools/intel.rst:57]: Yes, it works. For sure!

File: reference/tools/intel.rst (hunk 3 above)
Comment (franramirez688) [reply reference/tools/intel.rst:147]: I had an URL to go there, but Intel changes the domain too frequently, so I decided to remove it.

**Review by memsharded — APPROVED** [review]

**Review by czoido — APPROVED** [review]

**Pull request merged**